app/bundles/UserBundle/Entity/User.php

```diff
@@ -302,19 +302,19 @@ public static function loadApiMetadata(ApiMetadataDriver $metadata)
      */
     protected function isChanged($prop, $val)
     {
-        $getter  = 'get'.ucfirst($prop);
-        $current = $this->$getter();
         if ('role' == $prop) {
-            if ($current && !$val) {
-                $this->changes['role'] = [$current->getName().' ('.$current->getId().')', $val];
-            } elseif (!$this->role && $val) {
-                $this->changes['role'] = [$current, $val->getName().' ('.$val->getId().')'];
-            } elseif ($current && $val && $current->getId() != $val->getId()) {
-                $this->changes['role'] = [
-                    $current->getName().'('.$current->getId().')',
-                    $val->getName().'('.$val->getId().')',
-                ];
+            $current = $this->getRole();
+            if (!$current instanceof Role) {
+                return;
```

Comment:
@rahuld-dev Should we `return` or `throw` an exception?

Reply:
@shinde-rahul 👆